pkg/bgpv1/agent/controller.go

```diff
@@ -176,6 +196,18 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
 		return store.List(), nil
 	})
 
+	secrets, err := c.SecretResource.Store(startCtx)
```

Comment:
If we're only ever getting the secret from the cached store here once, should the underlying watcher continue running?

Would the intention be to continue to watch for updates and reload the secret?

Reply:
Yes, but the idea isn't to reload the secret, as if the BGP connection is working it doesn't make sense to tear it down just to update the password. Hence the watcher will update in the background and the secret may be fetched again on a new connection.